File: backend/src/main/java/com/gencel/backend/dto/CreateUserRequest.java
package com.gencel.backend.dto;

import com.gencel.backend.entity.User;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateUserRequest {

    @NotNull(message = "Institution ID is required")
    private UUID institutionId;

    @NotNull(message = "Role is required")
    private User.UserRole role;

    @NotBlank(message = "First name is required")
    private String firstName;

    @NotBlank(message = "Last name is required")
    private String lastName;

    @NotBlank(message = "Phone number is required")
    private String phoneNumber;

    @NotBlank(message = "Email is required")
    @Email(message = "Email should be valid")
    private String email;

    @NotBlank(message = "Password is required")
    @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).{8,}$", 
             message = "Password must be at least 8 characters long and contain at least one uppercase letter, one lowercase letter, and one digit")
    private String password;

    private String address;

    private Double latitude;
    private Double longitude;

    // Student specific field
    private String iban;
}

File: frontend/lib/models/create_user_request.dart
import 'package:tdp_frontend/models/user.dart';

class CreateUserRequest {
  final Role role;
  final String firstName;
  final String lastName;
  final String phoneNumber;
  final String email;
  final String password;
  final String? address;
  final double? latitude;
  final double? longitude;
  final String? iban;

  CreateUserRequest({
    required this.role,
    required this.firstName,
    required this.lastName,
    required this.phoneNumber,
    required this.email,
    required this.password,
    this.address,
    this.latitude,
    this.longitude,
    this.iban,
  });

  Map<String, dynamic> toJson() {
    return {
      'role': role.name,
      'firstName': firstName,
      'lastName': lastName,
      'phoneNumber': phoneNumber,
      'email': email,
      'password': password,
      if (address != null) 'address': address,
      if (latitude != null) 'latitude': latitude,
      if (longitude != null) 'longitude': longitude,
      if (iban != null) 'iban': iban,
    };
  }
}
